--- a/purpur-server/build.gradle.kts
+++ b/purpur-server/build.gradle.kts
@@ -36,12 +_,36 @@
             outputDir = rootDirectory.dir("paper-server")
         }
     }
-    activeFork = purpur
+    // activeFork = purpur // Nautilus
     // Purpur end - Rebrand
 
+    // Nautilus start - Rebrand
+    val nautilus = forks.register("nautilus") {
+        forks = purpur
+        upstream.patchRepo("paperServer") {
+            upstreamRepo = purpur.patchedRepo("paperServer")
+            patchesDir = rootDirectory.dir("nautilus-server/paper-patches")
+            outputDir = rootDirectory.dir("paper-server")
+        }
+        upstream.patchDir("purpurServer") {
+            upstreamPath = "purpur-server"
+            excludes = setOf(
+                "src/minecraft",
+                "paper-patches",
+                "minecraft-patches",
+                "build.gradle.kts",
+                "build.gradle.kts.patch"
+            )
+            patchesDir = rootDirectory.dir("nautilus-server/purpur-patches")
+            outputDir = rootDirectory.dir("purpur-server")
+        }
+    }
+    activeFork = nautilus
+    // Nautilus end - Rebrand
+
     spigot {
         enabled = true
-        buildDataRef = "436eac9815c211be1a2a6ca0702615f995e81c44"
+        buildDataRef = "62f9b85b300b6bbf1c559e379567699a6e281cec"
         packageVersion = "v1_21_R5" // also needs to be updated in MappingEnvironment
     }
 
@@ -125,10 +_,14 @@
     main {
         java { srcDir("../paper-server/src/main/java") }
         resources { srcDir("../paper-server/src/main/resources") }
+        java { srcDir("../purpur-server/src/main/java") }
+        resources { srcDir("../purpur-server/src/main/resources") }
     }
     test {
         java { srcDir("../paper-server/src/test/java") }
         resources { srcDir("../paper-server/src/test/resources") }
+        java { srcDir("../purpur-server/src/test/java") }
+        resources { srcDir("../purpur-server/src/test/resources") }
     }
 }
 val log4jPlugins = sourceSets.create("log4jPlugins") {
@@ -156,7 +_,7 @@
 }
 
 dependencies {
-    implementation(project(":purpur-api")) // Purpur
+    implementation(project(":nautilus-api")) // Purpur // Nautilus
     implementation("ca.spottedleaf:concurrentutil:0.0.3")
     implementation("org.jline:jline-terminal-ffm:3.27.1") // use ffm on java 22+
     implementation("org.jline:jline-terminal-jni:3.27.1") // fall back to jni on java 21
@@ -231,14 +_,14 @@
         val gitBranch = git.exec(providers, "rev-parse", "--abbrev-ref", "HEAD").get().trim()
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
-            "Implementation-Title" to "Purpur", // Purpur
+            "Implementation-Title" to "Nautilus", // Purpur // Nautilus
             "Implementation-Version" to implementationVersion,
             "Implementation-Vendor" to date,
-            "Specification-Title" to "Purpur", // Purpur
+            "Specification-Title" to "Nautilus", // Purpur // Nautilus
             "Specification-Version" to project.version,
-            "Specification-Vendor" to "Purpur Team", // Purpur
-            "Brand-Id" to "purpurmc:purpur", // Purpur
-            "Brand-Name" to "Purpur", // Purpur
+            "Specification-Vendor" to "Obydux", // Purpur // Nautilus
+            "Brand-Id" to "obydux:nautilus", // Purpur // Nautilus
+            "Brand-Name" to "Nautilus", // Purpur // Nautilus
             "Build-Number" to (build ?: ""),
             "Build-Time" to buildTime.toString(),
             "Git-Branch" to gitBranch,
@@ -257,7 +_,7 @@
 
 // Bump compile tasks to 1GB memory to avoid OOMs
 tasks.withType<JavaCompile>().configureEach {
-    options.forkOptions.memoryMaximumSize = "1G"
+    options.forkOptions.memoryMaximumSize = "4G" // Pufferfish
 }
 
 val scanJarForBadCalls by tasks.registering(io.papermc.paperweight.tasks.ScanJarForBadCalls::class) {
