--- a/purpur-api/build.gradle.kts
+++ b/purpur-api/build.gradle.kts
@@ -40,6 +_,7 @@
 }
 
 dependencies {
+    api("io.sentry:sentry:5.4.0") // Pufferfish
     // api dependencies are listed transitively to API consumers
     api("com.google.guava:guava:33.3.1-jre")
     api("com.google.code.gson:gson:2.11.0")
@@ -101,17 +_,21 @@
         java {
             srcDir(generatedDir)
             srcDir(file("../paper-api/src/main/java"))
+            srcDir(file("../purpur-api/src/main/java"))
         }
         resources {
             srcDir(file("../paper-api/src/main/resources"))
+            srcDir(file("../purpur-api/src/main/resources"))
         }
     }
     test {
         java {
             srcDir(file("../paper-api/src/test/java"))
+            srcDir(file("../purpur-api/src/test/java"))
         }
         resources {
             srcDir(file("../paper-api/src/test/resources"))
+            srcDir(file("../purpur-api/src/test/resources"))
         }
     }
 }
@@ -159,6 +_,13 @@
     }
 }
 
+// Pufferfish Start
+tasks.withType<JavaCompile> {
+        val compilerArgs = options.compilerArgs
+        compilerArgs.add("--add-modules=jdk.incubator.vector")
+    }
+// Pufferfish End
+
 tasks.jar {
     from(generateApiVersioningFile.map { it.outputs.files.singleFile }) {
         into("META-INF/maven/${project.group}/${project.name}")
@@ -179,6 +_,9 @@
 tasks.withType<Javadoc>().configureEach {
     (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:none", "-quiet") // Purpur - silence Paper's bajillion javadoc warnings
     val options = options as StandardJavadocDocletOptions
+    // Pufferfish start
+    options.addStringOption("-add-modules", "jdk.incubator.vector")
+    // Pufferfish end
     options.overview = "../paper-api/src/main/javadoc/overview.html"
     options.use()
     options.isDocFilesSubDirs = true
