--- a/purpur-api/build.gradle.kts
+++ b/purpur-api/build.gradle.kts
@@ -57,6 +_,7 @@
 }
 
 dependencies {
+    api("io.sentry:sentry:5.4.0") // Pufferfish
     // api dependencies are listed transitively to API consumers
     api("com.google.guava:guava:33.6.0-jre")
     api("com.google.code.gson:gson:2.14.0")
@@ -109,7 +_,7 @@
     testRuntimeOnly("org.junit.platform:junit-platform-launcher:6.0.3")
 
     // checkstyle
-    checkstyle(project(":purpur-checkstyle"))
+    //checkstyle(project(":purpur-checkstyle")) // Nautilus - Uncomment once this can be used in forks
 }
 
 val generatedDir: java.nio.file.Path = rootProject.layout.projectDirectory.dir("paper-api/src/generated/java").asFile.toPath() // Purpur
@@ -123,17 +_,21 @@
         java {
             srcDir(generatedDir)
             srcDir(file("../paper-api/src/main/java"))
+            srcDir(file("../purpur-api/src/main/java"))
         }
         resources {
             srcDir(file("../paper-api/src/main/resources"))
+            srcDir(file("../purpur-api/src/main/resources"))
         }
     }
     test {
         java {
             srcDir(file("../paper-api/src/test/java"))
+            srcDir(file("../purpur-api/src/test/java"))
         }
         resources {
             srcDir(file("../paper-api/src/test/resources"))
+            srcDir(file("../purpur-api/src/test/resources"))
         }
     }
 }
@@ -199,6 +_,13 @@
     apiVersion.set(rootProject.providers.gradleProperty("apiVersion"))
 }
 
+// Pufferfish Start
+tasks.withType<JavaCompile> {
+    val compilerArgs = options.compilerArgs
+    compilerArgs.add("--add-modules=jdk.incubator.vector")
+}
+// Pufferfish End
+
 tasks.jar {
     from(generateApiVersioningFile.flatMap { it.outputFile })
     manifest {
@@ -217,6 +_,7 @@
 tasks.withType<Javadoc>().configureEach {
     (options as StandardJavadocDocletOptions).addStringOption("Xdoclint:none", "-quiet") // Purpur - silence Paper's bajillion javadoc warnings
     val options = options as StandardJavadocDocletOptions
+    options.addStringOption("-add-modules", "jdk.incubator.vector") // Pufferfish
     options.overview = "../paper-api/src/main/javadoc/overview.html"
     options.use()
     options.isDocFilesSubDirs = true
