--- a/net/minecraft/world/entity/npc/villager/Villager.java
+++ b/net/minecraft/world/entity/npc/villager/Villager.java
@@ -393,7 +_,10 @@
         super.inactiveTick();
     }
     // Paper end - EAR 2
-
+    // Pufferfish start
+    public long nextGolemPanic = -1;
+    private int behaviorTick = 0;
+    // Pufferfish end
     @Override
     protected void customServerAiStep(ServerLevel level) {
         // Paper start - EAR 2
@@ -410,7 +_,7 @@
         } else {
             this.isLobotomized = false;
         }
-        if (!inactive && (getRider() == null || !this.isControllable())) { // Purpur - Ridables
+        if (!inactive && this.behaviorTick++ % this.activatedPriority == 0 && (getRider() == null || !this.isControllable())) { // Purpur - Ridables
             this.getBrain().tick(level, this); // Paper - EAR 2
         }
         else if (this.isLobotomized && shouldRestock(level)) restock();
