From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Obydux <obydux@proton.me>
Date: Wed, 9 Jul 2025 19:02:44 +0200
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index fb1fe2651e53a9bf46b3632c638e13eea9dcda93..cc2be4b6948ed5d26a8314787322b770c22eae84 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -25,6 +25,12 @@ public interface ServerBuildInfo {
      */
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
+    // Nautilus start
+    /**
+     * The brand id for Nautilus.
+     */
+    Key BRAND_NAUTILUS_ID = Key.key("obydux", "nautilus");
+    // Nautilus end
     /**
      * Gets the {@code ServerBuildInfo}.
      *
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 69c0ca1b0c0cdbc911ba3f053fafc5624539917d..7667b36872a0769d5bec9188e477ed47b3b94a96 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -2399,6 +2399,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Purpur end
 
+        // Nautilus start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getNautilusConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // Nautilus end
+
         /**
          * Sends the component to the player
          *
