--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -23,6 +_,13 @@
      */
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
+
+    // Nautilus start
+    /**
+     * The brand id for Nautilus.
+     */
+    Key BRAND_NAUTILUS_ID = Key.key("obydux", "nautilus");
+    // Nautilus end
     /**
      * Gets the {@code ServerBuildInfo}.
      *
