--- a/net/minecraft/world/entity/monster/Shulker.java
+++ b/net/minecraft/world/entity/monster/Shulker.java
@@ -401,7 +_,7 @@
 
     @Nullable
     protected Direction findAttachableSurface(BlockPos pos) {
-        for (Direction direction : Direction.values()) {
+        for (Direction direction : Direction.VALUES) { // Pluto - Avoid excessive calls to Enum#values
             if (this.canStayAt(pos, direction)) {
                 return direction;
             }
@@ -435,6 +_,7 @@
     }
 
     protected boolean teleportSomewhere() {
+        if (this.level().plutoConfig().disableShulkerTeleporting) return false; // Pluto - Shulker configuration
         if (!this.isNoAi() && this.isAlive()) {
             BlockPos blockPos = this.blockPosition();
 
@@ -515,6 +_,7 @@
         AABB boundingBox = this.getBoundingBox();
         // Purpur start - Shulker spawn from bullet options
         if ((!this.level().purpurConfig.shulkerSpawnFromBulletRequireOpenLid || !this.isClosed()) && this.teleportSomewhere()) {
+            if (this.level().plutoConfig().disableShulkerSplitting) return; // Pluto - Shulker configuration
             float chance = this.level().purpurConfig.shulkerSpawnFromBulletBaseChance;
             if (!this.level().purpurConfig.shulkerSpawnFromBulletNearbyEquation.isBlank()) {
                 int nearby = this.level().getEntities((net.minecraft.world.level.entity.EntityTypeTest) EntityType.SHULKER, boundingBox.inflate(this.level().purpurConfig.shulkerSpawnFromBulletNearbyRange), Entity::isAlive).size();
